app/helpers/application_helper.rb

```diff
@@ -475,11 +475,25 @@ def accessible_resource_records(resource, order = :name)
   end
 
   def accessible_resource(obj, resource, order = :name, association: resource)
-    list = accessible_resource_records(resource, order).to_a
+    klass = resource.to_s.classify.constantize
+
+    list = klass.unscoped.authorized
+
+    list = list.with_taxonomy_scope_override(@location, @organization) if klass.include? Taxonomix
+
     # we need to allow the current value even if it was filtered
-    current = obj.public_send(association) if obj.respond_to?(association)
-    list |= [current] if current.present?
-    list
+    if obj.respond_to?(association)
+      current = obj.public_send(association)
+      # match the unscope and readonly values from the left query
+      list = list.or(klass.unscoped.unscope(list.unscope_values).readonly(list.readonly_value).where(:id => current.id)) if current.present?
+    end
+
+    list.reorder(order)
+  end
+
+  def accessible_resource_for_select(obj, resource, order: :name, association: resource, columns: [:id, :name])
+    columns = Array(columns)
+    accessible_resource(obj, resource, order, association: association).pluck(*columns)
   end
 
   def accessible_related_resource(obj, relation, order: :name, where: nil)
```

Comment:
Nit: We should probably return `obj.none` (an empty active record result) instead of an empty array here.

Reply:
Not sure about this, if records are fetched this function returns an array of tuples (arrays), not ActiveRecord models, therefore it cannot be chained with other query methods. Returning an empty array ensures the same return type.
I can test the code with `obj.none` and check that everything still works.

Reply:
Are you sure? I thought it always returns an active record relation. Only the guard clause returns an empty array. And that's pretty inconsistent.

Reply:
According to the [Active Record Querying](https://guides.rubyonrails.org/active_record_querying.html#pluck) documentation the method `pluck` returns an array.

Reply:
Yeah, but we don't use `pluck` in `accessible_related_resource` :-D